src/hotspot/share/opto/stringopts.cpp

```diff
@@ -591,12 +556,127 @@ StringConcat* PhaseStringOpts::build_candidate(CallStaticJavaNode* call) {
   return NULL;
 }
 
+static bool is_string_startsWith(Node* call) {
+  if (call->is_CallStaticJava()) {
+    CallStaticJavaNode* csj = call->as_CallStaticJava();
+    ciMethod* m = csj->method();
+    if (m != NULL &&
+        m->intrinsic_id() == vmIntrinsics::_String_startsWith) {
+      return true;
+    }
+  }
+
+  return false;
+}
+
+void PhaseStringOpts::eliminate_call(CallNode* call, CallProjections& projs) {
```

Comment:
Similar comment here.  Could the eliminate_call() function after patch be in same sequence as before patch?